pandas/_libs/tslib.pyx

```diff
@@ -664,11 +664,11 @@ cpdef array_to_datetime(
 
                         # Still raise OutOfBoundsDatetime,
                         # as error message is informative.
-                        raise
+                        raise OutOfBoundsDatetime(f"Cannot convert \"{val}\" at position {i} to datetime")
```

Comment:
I find the `Out of bounds` messaging still important here. Could the position information be appended to the original message?